cmd/telemetrygen/pkg/traces/worker.go

```diff
@@ -35,15 +36,67 @@ type worker struct {
 	wg               *sync.WaitGroup       // notify when done
 	loadSize         int                   // desired minimum size in MB of string data for each generated trace
 	spanDuration     time.Duration         // duration of generated spans
+	numSpanLinks     int                   // number of span links to generate per span
 	logger           *zap.Logger
-	allowFailures    bool // whether to continue on export failures
+	allowFailures    bool                // whether to continue on export failures
+	spanContexts     []trace.SpanContext // collection of span contexts for linking
+	spanContextsMu   sync.RWMutex        // mutex for spanContexts slice
 }
 
 const (
-	fakeIP string = "1.2.3.4"
+	fakeIP                string = "1.2.3.4"
+	maxSpanContextsBuffer int    = 1000 // Maximum number of span contexts to keep for linking
 )
 
-func (w worker) simulateTraces(telemetryAttributes []attribute.KeyValue) {
+// addSpanContext safely adds a span context to the worker's collection
+// Maintains a circular buffer to prevent unbounded memory growth
+func (w *worker) addSpanContext(spanCtx trace.SpanContext) {
+	w.spanContextsMu.Lock()
+	defer w.spanContextsMu.Unlock()
+
+	w.spanContexts = append(w.spanContexts, spanCtx)
+
+	// Keep only the most recent span contexts to prevent memory growth
+	if len(w.spanContexts) > maxSpanContextsBuffer {
+		copy(w.spanContexts, w.spanContexts[len(w.spanContexts)-maxSpanContextsBuffer:])
+		w.spanContexts = w.spanContexts[:maxSpanContextsBuffer]
+	}
+}
+
+// generateSpanLinks creates span links to random existing span contexts
+func (w *worker) generateSpanLinks() []trace.Link {
+	if w.numSpanLinks <= 0 {
+		return nil
+	}
+
+	w.spanContextsMu.RLock()
+	defer w.spanContextsMu.RUnlock()
+
+	availableContexts := len(w.spanContexts)
+	if availableContexts == 0 {
+		return nil
+	}
+
+	links := make([]trace.Link, 0, w.numSpanLinks)
+
+	// Generate links to random existing span contexts
+	for i := 0; i < w.numSpanLinks && i < availableContexts; i++ {
```

Comment:
```suggestion
	for i := 0; i < w.numSpanLinks; i++ {
```

nit: currently, if e.g. 10 span links are requested but only 5 contexts are available, it's still possible that this code generates 5 span links linking to the same context (because the context is choosen at random)

I'd either make the code more complex to avoid this situation entirely (never link to the same span context twice in the same span), or ignore it (e.g. if the user asks for 10 span links but there is e.g. only 1 context, then we'll generate 10 span links, all linking to the same context).

Reply:
Removed the constraint to allow generating the requested number of span links even with
  fewer available contexts (allows duplicates)